content/1_General/Adding_Solution.mdx

```diff
@@ -88,24 +88,33 @@ See [Working With MDX](/general/working-mdx) for additional information.
     ```
 
     Keep file names and solution IDs consistent; namely, USACO IDs should
-    correspond to the URL on
+    correspond to the last digits at the very end of the URL on
     [usaco.org](http://www.usaco.org/index.php?page=viewproblem2&cpid=690) and
-    file names should be the same as the solution's ID.
+    file names should be the same as the solution's ID. See
+    [Working With MDX](/general/working-mdx#problem-lists) for more examples of
+    IDs.
 
 3.  Add your implementation, following these conventions and the conventions
     [below](#code-conventions):
 
-    - Use _tabs_ to indent.
-    - Lines should not exceed 80 characters.
+    - Use _tabs_ to indent
+    <Info title="Pro Tip">
+	If your code is written with spaces, use the [Guide Editor](https://usaco.guide/editor/), copy-paste your code in, right click, click "Command Palette" from the menu that pops up and then search for "Convert Indentation to Tabs". Once you see it, click on it and then copy that new code with the correct indentation. into your Github pull request.
+    </Info>
+    
+    - Lines should not exceed 80 characters. If you need to, you can write comments across multiple lines to follow this rule.
+    - Your code needs to have comments if it doesn't contribute anything to an already existing solution on the guide or official USACO solution. Comments shouldn't be used for every single obvious line and should only describe complicated or big chunks of code. Comments should be short and concise, being not longer than 2-3 sentences but still encompassing all the valuable information an inexperienced coder might not be able to automatically assume. 
     - Be _consistent_ about formatting your code, unlike the following snippet.
-      We don't care about which one you use (`a+b` or `a + b`), but pick one and
-      stick with it!
+      We don't care about which one you use (`a+b` or `a + b`), but pick one and stick with it throughout your entire code!
 
-      ```py
+     ```py
       a,b = (0,1)
       print(a+b)
       print(a + b)
       ```
+    - Use universally understandable variable and data structure names! Simply calling an integer that serves more purpose than just the input or output as "x" or "sus" will not suffice. Please use variable names that also make sense in respect to the problem that you are writing your solution to. 
+    - Remember, your pull request will not be accepted until every single aspect of your code is close to perfect, so please try to follow these rules to get it right the first time.
```

Comment:
I don't really know if this line is necessary? idk :shrug:

Reply:
it is because i didn't think that you guys would adhere to these guidelines so strictly so I went kinda loose with them and ended up having to redo most of my code multiple times because of it.

Reply:
can u change like the until it is almost perfect part. doesnt sound very nice xD

Reply:
idk I just don't see the point and I feel like this line gives off bad vibes 😛

Reply:
yea that's what i'm feeling too xd

Reply:
okay ill make it sound nicer and also more warm